File: Recursion/JavaScript/reverseString.js

let str = [...'hello']


function reverseStr(arr,i,j){
    if(i>j){
        return
    }
    const temp = arr[i]
    arr[i] = arr[j]
    arr[j] = temp
    reverseStr(arr,i+1,j-1)
}

reverseStr(str,0,str.length-1)

console.log(str.join(''))

File: Recursion/Python/reverse-string.py

st = list('hello')

def reverseString(arr,i,j):
    if i>j:
        return
    arr[i],arr[j] = arr[j],arr[i]
    reverseString(arr, i+1, j-1)

reverseString(st, 0, len(st)-1)

print(''.join(st))
